pkg/validation/ingestion_policies.go

```diff
@@ -143,3 +157,56 @@ func (p *PolicyStreamMapping) ApplyDefaultPolicyStreamMappings(defaults PolicySt
 	}
 	return nil
 }
+
+// policyContextKey is used as a key for context values to avoid collisions
+type policyContextKey int
+
+const (
+	ingestionPolicyContextKey policyContextKey = 1
+)
+
+// ExtractIngestionPolicyHTTP retrieves the ingestion policy from the HTTP header and returns it.
+// Returns the policy string and a boolean indicating whether the policy was found.
+func ExtractIngestionPolicyHTTP(r *http.Request) (string, bool) {
+	policy := r.Header.Get(HTTPHeaderIngestionPolicyKey)
+	if policy == "" {
+		return "", false
+	}
+	return policy, true
+}
+
+// InjectIngestionPolicyContext returns a derived context containing the provided ingestion policy.
+func InjectIngestionPolicyContext(ctx context.Context, policy string) context.Context {
+	return context.WithValue(ctx, any(ingestionPolicyContextKey), policy)
+}
+
+// ExtractIngestionPolicyContext gets the embedded ingestion policy from the context.
+// Returns the policy string and a boolean indicating whether the policy was found.
+func ExtractIngestionPolicyContext(ctx context.Context) (string, bool) {
+	policy, ok := ctx.Value(ingestionPolicyContextKey).(string)
+	return policy, ok
+}
+
+type ingestionPolicyMiddleware struct {
+	logger log.Logger
+}
+
+// NewIngestionPolicyMiddleware creates a middleware that extracts the ingestion policy
+// from the HTTP header and injects it into the context of the request.
+func NewIngestionPolicyMiddleware(logger log.Logger) middleware.Interface {
+	return &ingestionPolicyMiddleware{
+		logger: logger,
+	}
+}
+
+// Wrap implements the middleware interface
+func (m *ingestionPolicyMiddleware) Wrap(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		policy, isSet := ExtractIngestionPolicyHTTP(r)
```

Comment:
I see the bool as redundant. It's not possible to have an empty policy where `isSet` is also true because in the HTTP specification you cannot have an HTTP header with an empty value. I think the following would be simpler, but I'll leave the decision up to you 👍 


```suggestion
		policy := ExtractIngestionPolicyHTTP(r)
		if policy != "" {
```

Reply:
Ah I see you did this to make it have the same signature as `ExtractIngestionPolicyContext`. I'll leave the decision up to you.